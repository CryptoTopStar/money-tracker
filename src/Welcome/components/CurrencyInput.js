import React from 'react'

import { currencySymbol, currencyExponent } from '../../data/currency'
import { Input, Label } from 'semantic-ui-react'
import Cleave from 'cleave.js/react'

const CurrencyInput = ({ value, code, update }) => (
  <Input labelPosition="left" fluid>
    <Label>{currencySymbol(code)}</Label>
    <Cleave
      name={code}
      value={value || ''}
      options={{
        numeral: true,
        numeralDecimalScale: currencyExponent(code)
      }}
      onChange={e => update(code, e.target.rawValue)}
    />
  </Input>
)

export default CurrencyInput
