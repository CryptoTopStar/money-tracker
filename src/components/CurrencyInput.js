import React from 'react'
import Cleave from 'cleave.js/react'
import { Input, Label } from 'semantic-ui-react'
import { CURRENCY } from '../constants/currency'

const CurrencyInput = ({ value, code, onChange }) => (
  <Input labelPosition="left" fluid>
    <Label className="mono">
      {code}
    </Label>
    <Cleave
      name={code}
      value={value || ''}
      options={{
        numeral: true,
        numeralDecimalScale: CURRENCY[code].exp
      }}
      onChange={onChange}
      autoComplete={false}
    />
  </Input>
)

export default CurrencyInput
