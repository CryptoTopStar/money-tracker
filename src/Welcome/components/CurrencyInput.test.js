import React from 'react'
import renderer from 'react-test-renderer'
import CurrencyInput from './CurrencyInput'
global.navigator = {
  userAgent: 'node.js'
}

it('should render currency input', () => {
  const tree = renderer
    .create(
      <CurrencyInput
        value={100}
        code="USD"
        update={() => {}}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
