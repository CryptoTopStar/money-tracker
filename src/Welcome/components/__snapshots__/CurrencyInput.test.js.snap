// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render currency input 1`] = `
<div
  className="ui fluid left labeled input"
>
  <div
    className="ui label"
    onClick={[Function]}
  >
    $
  </div>
  <input
    data-cleave-ignore={
      Array [
        100,
        Object {
          "initValue": 100,
          "numeral": true,
          "numeralDecimalScale": 2,
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
    name="USD"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="100"
  />
</div>
`;
